### PR TITLE
SUMO-170407: initial support for new ServiceNow connection

### PR DESCRIPTION
**Pull request by rsanbhad**

### CODE REVIEW

File: website/docs/r/connection.html.markdown [hunk 1]
```diff
@@ -48,7 +48,8 @@ The following arguments are supported:
 - `headers` - (Optional) Map of access authorization headers.
 - `custom_headers` - (Optional) Map of custom webhook headers
 - `default_payload` - (Required) Default payload of the webhook.
-- `webhook_type` - (Optional) Type of webhook. Valid values are `AWSLambda`, `Azure`, `Datadog`, `HipChat`, `PagerDuty`, `Slack`, `Webhook`, `NewRelic`, and `MicrosoftTeams`. Default: `Webhook`
+- `connection_subtype` - (Optional) The subtype of the connection. Valid values are `Incident` and `Event`. NOTEL This is only used for the `ServiceNow` webhook type.
```
Comment (kumar-avijit) [inline website/docs/r/connection.html.markdown:51]: fix typo? `NOTEL` -> `NOTE:`

File: sumologic/resource_sumologic_connection.go [hunk 2]
```diff
@@ -77,9 +77,14 @@ func resourceSumologicConnection() *schema.Resource {
...
 				Type:         schema.TypeString,
 				Optional:     true,
-				ValidateFunc: validation.StringInSlice([]string{"AWSLambda", "Azure", "Datadog", "HipChat", "PagerDuty", "Slack", "Webhook", "NewRelic", "Jira", "Opsgenie", "MicrosoftTeams"}, false),
+				ValidateFunc: validation.StringInSlice([]string{"AWSLambda", "Azure", "Datadog", "HipChat", "PagerDuty", "Slack", "Webhook", "NewRelic", "Jira", "Opsgenie", "MicrosoftTeams", "ServiceNow"}, false),
 				Default:      "Webhook",
 			},
+			"connection_subtype": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Incident", "Event", ""}, false),
```
Comment (kumar-avijit) [inline sumologic/resource_sumologic_connection.go:86]: I wonder what's the best practice around writing empty arguments (`""`) in plans vs omitting them 🤔 
Would it break anything if we disallowed empty `connection_subtype`?

**Review by kumar-avijit — APPROVED** [review]
couple of comments, but lgtm

File: sumologic/resource_sumologic_connection.go (hunk 2 above)
Comment (rsanbhad) [reply sumologic/resource_sumologic_connection.go:86]: ah yup empty string would be invalid on server-side - I think I added the empty string while trying to figure out why the tests are failing (it was a different issue), will remove it

**Pull request merged**